textdb/textdb-common/src/main/java/edu/uci/ics/textdb/common/field/Span.java

```diff
@@ -16,4 +40,36 @@ public int getStart() {
     public int getEnd() {
         return end;
     }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + end;
+        result = prime * result
+                + ((fieldName == null) ? 0 : fieldName.hashCode());
+        result = prime * result + start;
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+        Span other = (Span) obj;
+        if (end != other.end)
+            return false;
+        if (fieldName == null) {
+            if (other.fieldName != null)
+                return false;
+        } else if (!fieldName.equals(other.fieldName))
+            return false;
+        if (start != other.start)
+            return false;
+        return true;

```

Comment:
@chenlica .. I initially thought span equality shouldn't depend on the key/value, since span is characteristic of start, begin and field. But on second thought, I feel they should also be included since there can be same spans (same start, end, field) with different key. 
I will include the other field key and value